backend/src/interface/persistent_data.rs

```diff
@@ -26,77 +26,91 @@ pub enum DataError {
 }
 
 #[async_trait]
-/// An interface for checking relations and inserting data structures. The MealplanManagement component uses this interface for database access.
+/// An interface for checking relations and inserting data structures. The MealPlanManagement component uses this interface for database access.
 pub trait MealplanManagementDataAccess: Send + Sync {
     /// Removes all relations to the meal plan at the given date and the given canteen.
     /// Without removing changes in the meal plan couldn't be updated.
-    async fn dissolve_relations(&self, canteen: Canteen, date: Date) -> Result<()>;
+    async fn dissolve_relations(&self, canteen_id: Uuid, date: Date) -> Result<()>;
     /// Determines the canteen with the most similar name.
-    async fn get_similar_canteen(&self, similar_name: &str) -> Result<Option<Canteen>>;
+    /// Returns the uuid to the similar canteen.
+    async fn get_similar_canteen(&self, similar_name: &str) -> Result<Option<Uuid>>;
     /// Determines the line with the most similar name.
-    async fn get_similar_line(&self, similar_name: &str) -> Result<Option<Line>>;
+    /// Returns the uuid to the similar line.
+    async fn get_similar_line(&self, similar_name: &str) -> Result<Option<Uuid>>;
     /// Determines the meal with the most similar name, identical allergens and identical additives.
+    /// Returns the uuid to the similar meal.
     async fn get_similar_meal(
         &self,
         similar_name: &str,
         allergens: &[Allergen],
         additives: &[Additive],
-    ) -> Result<Option<Meal>>;
+    ) -> Result<Option<Uuid>>;
     /// Determines the side with the most similar name, identical allergens and identical additives.
+    /// Returns the uuid to the similar canteen.
     async fn get_similar_side(
         &self,
         similar_name: &str,
         allergens: &[Allergen],
         additives: &[Additive],
-    ) -> Result<Option<Side>>;
+    ) -> Result<Option<Uuid>>;
 
-    /// Updates an existing canteen entity in the database. Returns the entity.
-    async fn update_canteen(&self, uuid: Uuid, name: &str) -> Result<Canteen>;
-    /// Updates an existing line entity in the database. Returns the entity.
-    async fn update_line(&self, uuid: Uuid, name: &str) -> Result<Line>;
-    /// Updates an existing meal entity in the database. Returns the entity.
-    async fn update_meal(
+    /// Updates an existing canteen entity in the database.
+    /// Returns the canteen uuid.
+    async fn update_canteen(&self, uuid: Uuid, name: &str) -> Result<Uuid>;
+    /// Updates an existing line entity in the database.
+    /// Returns the line uuid.
+    async fn update_line(&self, uuid: Uuid, name: &str) -> Result<Uuid>;
+    /// Updates an existing meal entity in the database.
+    async fn update_meal(&self, uuid: Uuid, name: &str) -> Result<()>;
+    /// Adds a meal into the meal plan with the given params.
+    async fn add_meal_to_plan(
```

Comment:
to add a meal to the mealplan the following (and nothing more) is neccesary:
- meal_id (food_id)
- line_id
- date
- price

Reply:
so exchange canteen for line